components/admin_console/permission_schemes_settings/edit_post_time_limit_modal/edit_post_time_limit_modal.tsx

```diff
@@ -0,0 +1,183 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useState} from 'react';
+import {FormattedMessage} from 'react-intl';
+import {Modal} from 'react-bootstrap';
+
+import {AdminConfig} from 'mattermost-redux/types/config';
+import {ActionFunc} from 'mattermost-redux/types/actions';
+
+import FormattedMarkdownMessage from 'components/formatted_markdown_message';
+import {Constants} from 'utils/constants';
+import {localizeMessage} from 'utils/utils.jsx';
+import {t} from 'utils/i18n';
+
+const INT32_MAX = 2147483647;
+
+type Props ={
+    config: Partial<AdminConfig>;
+    show: boolean;
+    onClose: () => void;
+    actions: {
+        updateConfig: (config: AdminConfig) => ActionFunc & {error?: ClientErrorPlaceholder};
+    };
+}
+
+type ClientErrorPlaceholder = {
+    message: string;
+    server_error_id: string;
+}
+
+export default function EditPostTimeLimitModal(props: Props) {
+    const {ServiceSettings} = props.config;
+
+    const [saving, setSaving] = useState(false);
+    const [errorMessage, setErrorMessage] = useState('');
+    const [postEditTimeLimit, setPostEditTimeLimit] = useState<number>(ServiceSettings?.PostEditTimeLimit || -1);
+    const [alwaysAllowPostEditing, setAlwaysAllowPostEditing] = useState(postEditTimeLimit > 0);
+
+    const save = async () => {
+        setSaving(true);
+        setErrorMessage('');
+
+        if (postEditTimeLimit.toString() === 'NaN' || postEditTimeLimit < -1 || postEditTimeLimit > INT32_MAX) {
+            setErrorMessage(localizeMessage('edit_post.time_limit_modal.invalid_time_limit', 'Invalid time limit'));
+            setSaving(false);
+            return false;
+        }
+
+        const newConfig = JSON.parse(JSON.stringify(props.config));
+        newConfig.ServiceSettings.PostEditTimeLimit = postEditTimeLimit;
+
+        const {error} = await props.actions.updateConfig(newConfig);
+        if (error) {
+            setErrorMessage(error.message);
+            setSaving(false);
+        } else {
+            setSaving(false);
+            props.onClose();
+        }
+
+        return true;
+    };
+
+    const handleOptionChange = ({currentTarget}: React.FormEvent<HTMLInputElement>) => {
+        setAlwaysAllowPostEditing(currentTarget.value === Constants.ALLOW_EDIT_POST_ALWAYS);
+        setPostEditTimeLimit(currentTarget.value === Constants.ALLOW_EDIT_POST_ALWAYS ? -1 : 0);
```

Comment:
I would not do that ... it is a bit of an improvement in usability, so whenever you change the value for `alwaysAllowPostEditing` the input field will be disabled, but the value stored in the state for it does not reset, so when the admin changes the radio button again it will go back to the value that was stored before and the admin does not have to set a new value for it.

That's basically the reasoning behind my original change request. :D

```suggestion
```

Reply:
no, thats not what I meant as well ..  we both might need to make ourselves clearer! :D

I would handle this inside the component without the need of updating the redux store/state.
One thing I missed though is when `alwaysAllowPostEditing` you would need to pass `-1` instead of `postEditTimeLimit` into the saving (`updateConfig`) action.

I will add a suggestion to line 53 as well